### PR TITLE
Add release notes for 0.230

### PR DESCRIPTION
**Pull request by caithagoras**
# Missing Release Notes
## Ami Tavory
- [x] https://github.com/prestodb/presto/pull/13574 Refactor Differential Entropy for Mutual Information Classification (Merged by: Rebecca Schlussel)

## Andrii Rosa
- [x] 621eda421c851f5fc126a84be71afd3737b370d6 Set boolean value to 0 if the position is null
- [x] c4462915ae5a5ee4d6009a3bd84c00933060e8b8 Check aggregation mask values for null

## Gaurav Mittal
- [x] https://github.com/prestodb/presto/pull/13704 Add support for JNI based decompression for zstd files (Merged by: James Sun)
- [x] https://github.com/prestodb/presto/pull/13697 Wrap OrcReader parameters into OrcReaderOptions (Merged by: James Sun)

## Leiqing Cai
- [x] ec9f4778b726128dc1b14a72efb9ce7cd8774856 Disable TestDistributedSpilledQueries#testLimitWithJoin
- [x] f189ca7dabb0044725bd59ab395b5152ffebe332 Make TestUpgradeMetadata single threaded

## Masha Basmanova
- [x] 08dba521159b03f98d803dbf6b55e828906254d9 Fix selective stream readers
- [x] 20653eabb90fee53d455dd18fce40dafc8702d9e Fix BooleanInputStream#getSetBits
- [x] c4d0c7f3fd1b11ac90f4dce9484f6685424caa02 Fix pushdown for filters on multiple subfields of a struct
- [x] 9d60d38328a2b4e4ef2eff3b31fdfda452dcfce7 Create NestedField using lowercase field name
- [x] 2b6e9749629bd41d4beff6809a050761e2a43f18 Make subfield name matching case insensitive

## Sahar Massachi
- [x] https://github.com/mayunlei/presto/pull/2 merge 

## Saksham Sachdev
- [x] https://github.com/prestodb/presto/pull/13660 Transaction Sharing across Connectors (Merged by: James Sun)

# Extracted Release Notes
- #13283 (Author: Laila A. Wahedi): imported combinations function from prestosql, complete with tests an…
  - Added combinations function, a function that returns n combinations of values in an array, up to n=5.
- #13405 (Author: Venki Korukanti): Reduce the identifier length in generated code for nested columns
  - Fix compilation errors for expressions over types containing an extremely large number of nested types.
- #13412 (Author: Leiqing Cai): Support DROP FUNCTION
  - Add support for ``DROP FUNCTION``.
- #13601 (Author: Islam Ismailov): Modularize Hive Metastore
  - Metastore interface is separated into a separate module to reduce monolithicness.
- #13602 (Author: Ying Su): Improve BatchStreamReader - Part 1
  - Improve BatchStreamReader performance.
- #13616 (Author: James A. Gill): Add geospatial expand_envelope function.
  - Add expand_envelope function to return a geometry's envelope expanded by a distance.
- #13644 (Author: James Sun): Introduce caching file system
  - Introduced new cache module to allow using local disk for to cache files from remote file systems.
  - Allow Raptor to read data from HDFS while caching the files on local disks.
- #13645 (Author: James A. Gill): Fix GeometryToBingTiles for certain geometries
  - Fix geometry_to_bing_tile for geometries at -180 longitude or 85.05112878 latitude.
- #13647 (Author: Leiqing Cai): Output verifier events for queries being skipped by pre-filters
  - Add skipped verification results to the output for queries being filtered.
- #13650 (Author: Yi He): Optimize lambda body when possible
  - Fix regression on lambda evaluation (Issue#13648).
- #13651 (Author: Shixuan Fan): Make commitPartition an async operation
  - Change ``ConnectorMetadata#commitPartition`` into async operation, and rename it to ``ConnectorMetadata#commitPartitionAsync``.
- #13652 (Author: Ke Wang): fix organizationDiscoveryIntervalMillis bug
  - Fix organizer not running scheduled job due to misreading `storage.organization-discovery-interval` config.
- #13653 (Author: Yi He): Fix PRESTO_EXTRA_CREDENTIAL parsing
  - Improved PRESTO_EXTRA_CREDENTIAL header parsing to allow value contain multiple '=' and urlEncode characters.
- #13734 (Author: Wenlei Xie): Add all_match, any_match and none_match functions for arrays
  - Add all_match(), any_match(), and none_match() functions.
- #13741 (Author: Ke Wang): Enable creating table with table_supports_delta_delete property for raptor
  - Add table_supports_delta_delete property in Raptor to allow deletion happening in background. DELETE queries in Raptor can now delete data logically but relying on compactors to delete physical data.
- #13743 (Author: Bin Fan): Add Alluxio client jar to hive-hadoop2
  - Add an Alluxio client jar to `plugin/hive-hadoop2/` (as a runtime dependency) to avoid copying Alluxio client jar to all Presto servers manually to connect to Alluxio.
- #13748 (Author: Rongrong Zhong): Fix SQL function compilation failure
  - Fix SQL function compilation failure when the function parameter is not referenced in function body.
- #13764 (Author: Rongrong Zhong): Respect null-call clause in SQL function execution
  - Fix SQL function execution not respecting null-call clause.
- #13784 (Author: Rongrong Zhong): Cast constants in SQL function to proper types
  - Fix SQL function byte code compilation error for functions implemented with constants.
- #13787 (Author: Leiqing Cai): List non built-in functions in SHOW FUNCTIONS
  - Add support to list non-builtin functions in SHOW FUNCTIONS. The feature can be turned on by the configuration property ``list-non-built-in-functions``.

# All Commits
- ec9f4778b726128dc1b14a72efb9ce7cd8774856 Disable TestDistributedSpilledQueries#testLimitWithJoin (Leiqing Cai)
- 08dba521159b03f98d803dbf6b55e828906254d9 Fix selective stream readers (Masha Basmanova)
- 621eda421c851f5fc126a84be71afd3737b370d6 Set boolean value to 0 if the position is null (Andrii Rosa)
- c4462915ae5a5ee4d6009a3bd84c00933060e8b8 Check aggregation mask values for null (Andrii Rosa)
- 20653eabb90fee53d455dd18fce40dafc8702d9e Fix BooleanInputStream#getSetBits (Masha Basmanova)
- c4d0c7f3fd1b11ac90f4dce9484f6685424caa02 Fix pushdown for filters on multiple subfields of a struct (Masha Basmanova)
- f189ca7dabb0044725bd59ab395b5152ffebe332 Make TestUpgradeMetadata single threaded (Leiqing Cai)
- 9d60d38328a2b4e4ef2eff3b31fdfda452dcfce7 Create NestedField using lowercase field name (Masha Basmanova)
- 2b6e9749629bd41d4beff6809a050761e2a43f18 Make subfield name matching case insensitive (Masha Basmanova)
- c5c3c670f3ba1552752740473fb07355ac52a2ae Use LazyBlock for columns with no filters (Masha Basmanova)
- 23e47735069360373b25e6604e460dec38353ce4 Add Alluxio client jar to hive-hadoop2 (Bin Fan)
- 2b7baf23d57ca858be569d7a69d6a2054c29eb4a List non built-in functions in SHOW FUNCTIONS (Leiqing Cai)
- 7236e03b2806dc0ace1b6a522106b123fff24886 Cast constants in SQL function to proper types (Rongrong Zhong)
- b35550cb99c994239aa978054961555cb90a6cd2 Remove some usages of QualifiedFunctionName#of(String) (Leiqing Cai)
- cb1b514a2a2e2dabbe9e9000f736702a78802973 Throw SYNTAX_ERROR when creating function with invalid name (Leiqing Cai)
- 943b4d0dddd266eb9615146e96bdd29f535b7ba2 Support DROP FUNCTION execution (Leiqing Cai)
- 71a23da4413adb97b26a132f085a00dbb7259328 Support dropFunction in FunctionNamespaceManager (Leiqing Cai)
- 8138d2f2b1ca83d0ef90b2ebeda8a6e4e83c0a57 Add syntax support for DROP FUNCTION (Leiqing Cai)
- eb47fcfe1121e82155dede9cb543156453c96f9d Remove print statement in TestFunctionNamespaceManager (Leiqing Cai)
- 7c26b6e0fd5763a542e8c1062a7e80e4d9310f67 Invert if condition for better readability (Ajay George)
- 1fbfb8e1c2d3869bee8fa8d25b86da77f6de3363 Add support for column names with slashes to SubfieldTokenizer (Masha Basmanova)
- 0f1fe83486c8fa8f1377dbf7480dc4c9587a2389 Fix backward compatible issue of Raptor delta delete (Ke Wang)
- d254d5f9464ada40b976653d8ae2b6fcc6997d01 Fix materialized exchange planning with filter pushdown (Masha Basmanova)
- df0fd4def8d04c4a7abdbb8ad64c978e3af913f0 Fix slice size overflow in DictionaryBlock.getIds() (Ying Su)
- 80613f9cbedd0fbd8ca08580ea56eee51ac2be55 Add support for column names with hyphen to SubfieldTokenizer (Masha Basmanova)
- 3242715959a169dbcdd88946c28488d2365c8886 Respect null-call clause in SQL function execution (Rongrong Zhong)
- 9aa8f80edf946d26760012d9da4c1b34cff25eb7 Optimize lambda body when possible (Yi He)
- 83b7d178f0513468593c7639f70d3751a83b8e87 Propagate correct error code in RowExpressionInterpreter (Yi He)
- bf4f7ec30be0f19e68edf768396a268d4f6cbad3 introduce warnings for nonReserved words moving to reserved (Sujay Jain)
- 5772869a17e518c48400224667d773cac058a4d8 Materialize stripe stream slice when caching (Shixuan Fan)
- 98f54e771a617a9a2c5fbd3ce70e65501e6b7964 Fix stripe metadata cache weigher to use retained size (Shixuan Fan)
- a472c23662f01bcf60607e11bad369a45a0b2d5c Fix filter pushdown for always-true range filter on a boolean column (Bhavani Hari)
- bc2e8b242da27e9b966035263f2d8516ed4ee731 chunk merge files during cache write to avoid reading excessively large data into memory and other optimizations (Sujay Jain)
- dcb3e1bb75c5521e9a5310b861209d5ad8e461fe Construct QualifiedFunctionName with CatalogSchemaName and function name (Vic Zhang)
- 5985800d12ae0f8cbdf08dbba7b54f1e8de67bfd Add array combinations function (Laila A. Wahedi)
- 36ec9d22557f8873c5b3145745ab63702d07a977 Enable creating table with table_supports_delta_delete property (Ke Wang)
- d355b8b596d2132ed02b8fb54307ffa7998c187e Add support for filter functions on columns with coercions (Bhavani Hari)
- 6dc5468bdf2ddb02a077646fb52f9a845b9e17bf Fix SQL function compilation failure (Rongrong Zhong)
- dedea3b46212ae9e0b00ee0fad446d75df70ace0 Support supplying benchmark queries through MySQL (Nikhil Collooru)
- 456487eeb5ac5d2b53d9edb81cd2fb25a046c4f2 Minor refactor of ApplyNode constructor (Vic Zhang)
- 19339ee0410f37bf27af9a4ad55d668024ad30f6 Throw PrestoException for unsupported correlated subquery (Vic Zhang)
- bd08887707f6201c3ac63876191f60011059558b Use distinct table names in TestHiveRecoverableGroupedExecution (Shixuan Fan)
- e18bb835ab1ac1f4a67f2fb2acd8bd74138d0d46 Expose H2QueryRunner Handle for Modular Testing (Saksham Sachdev)
- 0a0a4436a1b0bdb3b3c8daeed366d98744dc3859 Add support for Companion Catalogs in Transactions (Saksham Sachdev)
- 57f9632a9114bfd75762d8bbcdb2f4e5d203588e Revert "Optimize PlanFragment serialization" (Nikhil Collooru)
- cafa9e28d32740ed9656e40f0984956455b112bd Minor fixes to documents (Wenlei Xie)
- 4eb854909d8dcd245bd8f4af898c65bd302d6ff8 Rename session variable to be more specific (Bhavani Hari)
- 042634096c3fe4f660ee9e3ed225e037b4c34bdf Fix size type when calculating retained size. (Ying Su)
- 48e58fde4298acdb36b862aab21f2b56037bd084 Make restProxyServiceForQuery nullable for the Pinot connector (Devesh Agrawal)
- fca4353c5b4649cac9d92ad720a249e1e232840f Add all_match, any_match and none_match functions for arrays (Wenlei Xie)
- 2bee126caf9901eb25c621c5da0b5753f5d880df Extract and rename functional interfaces in ArrayFilterFunction (Wenlei Xie)
- df7a0e581b27afcef9d9fbe5a1f19d431fcec964 Remove unused FilterVoidLambda interface in ArrayFilterFunction (Wenlei Xie)
- 7a5cc80d614cd320c8b60ca98eb2ad8d56d169f4 Fix minor typo in CLA instructions (Brian Warner)
- bba36ec71f455b557579a17448d7330ad4597a79 Update CLA information to reflect EasyCLA (Brian Warner)
- ef3456a7a5239bc94b0599c03e19bf55be1801fc Introduce caching file system (James Sun)
- 229b60b9c76630b90b8142e15edb066ec3df7657 Make commitPartition an async operation (Shixuan Fan)
- 3d2f2a13a4d34e2ecc3e6166bbd088a3e1bd4a87 Add JNI decompressor for zstd files (Gaurav Mittal)
- 01c66c7959bcc1970d7761f3c54848f64b058605 Fix selective struct reader to make a copy of input positions (Bhavani Hari)
- 8e5051cd81afed8df74d697d4fc19770ea5952d1 Wrap OrcReader parameters into OrcReaderOptions (Gaurav Mittal)
- 70308fa48f7dd8a1b9ff1f6b62c51cd0f4d56769 Fix Values are too large error message (Masha Basmanova)
- cbf1ee762564fb1e5a33e2a978140f28141cd67a Allow range filters on newly added struct fields (Masha Basmanova)
- 998db977c233803322252035398c4251082a37c3 Add test for range filter on a newly added column (Masha Basmanova)
- d294ebca931a68155a06b090fddf5011c36c4050 Allow range filters on subfields of structs with coercion (Masha Basmanova)
- e668b1a526e013f4003c4947d1d79067e52d78e9 Reset allNulls flag in SliceDirectxxx#read (Masha Basmanova)
- db8067e93766014434ea9d5a8a95562726063150 Reset allNulls flag in SliceDictxxx#read (Masha Basmanova)
- c0fdc903fee99d2148ccdf417356ccfbfd60c0b8 Skip map entries with null keys (Masha Basmanova)
- d9e9f391ad06e6a59c9f111d9d0b3bd97eee7236 Run filters without inputs first in OrcSelectiveStreamReader (Sahar Massachi)
- c0b0edd2c81d1245a9f3000d4a9b382ea77344a8 Evalute predicate on partition columns to prune partitions (Masha Basmanova)
- dd16d8a486637da64dd8c4ab6df78063636967f3 Add a method to interpret an expression to ExpressionOptimizer (Masha Basmanova)
- ca4e94cbbf6e1091a119a04523d0eb46a24afe60 Add functionMetadataManager to HiveMetadataFactory (Masha Basmanova)
- 39a16b360f25e569f5aad49ffc4193408ad66a17 Fix filter pushdown for DELETE queries (Masha Basmanova)
- 6cced01a5b3e00a02e571f37b8ea5df21486df95 Support reading RC, TEXTFILE fromats with pushdown enabled (Bhavani Hari)
- d3ee8ec9d5da58eb8f2a24df1ce8955b434da549 Replace Joda-Time libraries with java.time (Ajay George)
- 315868da110434b23180dcf7fece3ae47589dd7c Add final to histogram class members (Ami Tavory)
- 9f2359a7b1487b09081045e3582e0e0048071fd2 Fixe minor issues in reservoir sample tests (Ami Tavory)
- 4b112927ceed96b7e635a30024506f75320edfa3 Move logic for handling different strategy implementations to DifferentialEntropyStateStrategy (Ami Tavory)
- 62299ab90df05e0ed0ecdfb8b557dcb80cc3a4ac Add cloneEmpty method to differential entropy strategy classes (Ami Tavory)
- 35308314d18e31a86e06973b1eee123f3eba2879 Add population weight/count methods to reservoir samples (Ami Tavory)
- 8e15a3afe5e0b7a294a5c251e0d8ac6d583aefee Expand entropy utils class and rename methods (Ami Tavory)
- 9b0a1a0c2a12588486d1aa15676ad2965ee57176 Update names of variables in code, and function letters in docs. (Ami Tavory)
- dacfc4519bd14b3319d5b8bb44e8b3f87c981b92 Remove unused types from CostCalculator (James Sun)
- 6cc6e5b83b97ba276484667a7db27be9869cec2c Fix the build broken by 9633991927 (Ying Su)
- 906006b09586aede02b33fe4fbc266a53574f2a2 Add memory tracking from StreamReader local buffers (Ying Su)
- 7be0c9ff4ea8808f7dbb637f55558fe48835b097 Improve BenchmarkBatchStreamReaders (Ying Su)
- f92fa2570639816e7be432869a6dde750480a487 Improve ORC LongReader (Ying Su)
- 61408c230dd96db0d40b9c235e28b5766bdbe1ab Improve ORC timestamp reader (Ying Su)
- 0814f331e7b7eeacf2cb2bc7c23bb319b1f4377d Improve ORC list and map readers (Ying Su)
- bf96a44cc82024b979c65df44b3312a47b8aebd5 Improve ORC slice direct reader (Ying Su)
- 9633991927fa30a15a6fc839671d57b196904364 Improve ORC byte reader (Ying Su)
- 95a57bd66e50253def785a535ca62dc3c9ff899f Improve ORC boolean reader (Ying Su)
- d45be17905be81c751b41c81865a2398b8031f29 Remove type from OrcRecordReader.readBlock (Ying Su)
- 96bdef063a49365b56eebb96fe2ffeb380b7df49 Add support for bucket adaptation (Masha Basmanova)
- 97bd9458f85ba88c0abbbcfc387e32ca6e621f4b Fix getTableStatistics with subfield pruning (Masha Basmanova)
- fb2079e7851c989b3636a08f488ae49562537f3a Clarify variable names in getTableStatistics (Masha Basmanova)
- 9805be6eae1ae81e4f947943b58e4e7f93cd5e06 Refactor FilterStatsCalculatorService (Masha Basmanova)
- c851382753821ec59c338881c17ec2975cd77f0a Fix subfield pruning for legacy unnest (Masha Basmanova)
- 53a84b0c0197067a4bdc893e0986d046378b94a0 Add test to verify the size of all-null blocks returned (Bhavani Hari)
- bd306cb9d7d5afc1ec6399cc36efb74a253743a6 Fix the size of all-null blocks returned from Decimal reader (Bhavani Hari)
- 7ae9f43fa5a32056b47aa3e9697dcdc783b0d12c Fix the size of all-null blocks returned from MapFlat reader (Bhavani Hari)
- 5b294b0d5a50c092939a5e62f4e08489c9da9296 Fix the size of all-null blocks returned from Timestamp reader (Bhavani Hari)
- b835ac11bc88feb8062777a62ca956aec5b25d2f Fix the size of all-null blocks returned from List reader (Bhavani Hari)
- 1b5d86dcaee5aed849fe7efcf0a108b3f70402fa Fix the size of all-null blocks returned from Slice reader (Bhavani Hari)
- 874255dfa980cb4374a2ecb3d7372c001101436d Fix the size of all-null blocks returned from Long reader (Bhavani Hari)
- 6d4250a3a373d06607482c5bb091092cadb702ff Fix the size of all-null blocks returned from Double reader (Bhavani Hari)
- 68ca8a606f81c43325246c6f109d420262e0fb23 Fix the size of all-null blocks returned from Float reader (Bhavani Hari)
- 300ba88daa74665cf88d373b0d62354443279b97 Fix the size of all-null blocks returned from Byte reader (Bhavani Hari)
- 1b6d7f97535bee4dd43006961c06b4608015a644 Fix the size of all-null blocks returned from Boolean reader (Bhavani Hari)
- 91af80f36d3dbc7331e262134c66a57b41b9bd49 Enable testGroupByKeyPredicatePushdown with filter pushdown (Masha Basmanova)
- 4e71bf3dc95c67396a345f19195d6e3c716ea528 Optimize byte reader when reading contiguous rows with no nulls and no filter (Ying Su)
- 4e66289e141ba344dcb9a5315e3dabc04ae819ea Test filter-only columns with coercions (Masha Basmanova)
- 604b7070f255f757c27a51c731f57f35d6cee373 Allow range filters on columns with coercion (Masha Basmanova)
- f5e14e9e721eedfb191d83d67f310389ef55f7f6 Static import TupleDomain.withColumnDomains (Masha Basmanova)
- 7e19193a8439f6e508391c54e76a2e798ad568f6 Add coercing tuple domain filters (Masha Basmanova)
- 5b874b4589bf80c4e9bac1297a1a4737c6a2eeee Introduce HiveCoercer interface (Masha Basmanova)
- 9c21aaa659f9d4927fe12bec4b8015c8cbf4722e Move out the entire Hive metastore (James Sun)
- b7dcdb9cf2f2ddcb3fbf86b88770de6da053353c Add requireNonNull checks to ConnectorManager (Sahar Massachi)
- 8f1b1ae6c2a0ed8819d2f22f6defbc143be85aa4 Incorporate pushed down filter into stats used by CBO (Sahar Massachi)
- 272a0e97c72a349e6ebce7c4290f116b116cc502 Rename toSymbolStatistics to toVariableStatsEstimate (Sahar Massachi)
- 154c4465dc1b48a9abcb47d58241fa9459f4c1ef Extract toSymbolStatistics from TableScanStatsRule (Sahar Massachi)
- 04019d48a374c2c6f4c6d52aef56ec6af1cb865b Add layout handle parameter to ConnectorMetadata.getTableStatistics (Masha Basmanova)
- abfef5022e864fa91c54e258e896f1cc74aa4fbc Change DomainTranslator.toPredicate to allow any RowExpression (Sahar Massachi)
- 282a7f17610c7ce34838228d49fce0e65867da9e Add Subfield to RowExpression converter (Sahar Massachi)
- 35970d38e0079d015b759a67cffb984ca9e88361 Fix filter pushdown for non-partitioned tables (Masha Basmanova)
- ed163684080a6fdb6c486f85bd8aebbc107c07de Static import Domain.singleValue (Masha Basmanova)
- 669f9b5558b03705f2ed2ba40bfc6637cd411bbe Move ExtendedHiveMetastore with its deps to new presto-hive-metastore module (Islam Ismailov)
- 1a18eb9fca7081dbd23d9aa573d746091c801ea3 Do not prune subfields when filter uses entire column (Masha Basmanova)
- d2a31579e6f68fc2a23c9c6837f075d6b3d4c424 Add basic support for schema evolution to OrcSelectiveRecordReader (Masha Basmanova)
- 685475f6678a3ccfdb02aebd0dd5a77d5ef6f3f8 Extract coercers from HivePageSource (Masha Basmanova)
- 7c61a7ee1c468872e7ed44f63052ca5bf916bc0f Add TestHiveClientInMemoryMetastoreWithFilterPushdown (Masha Basmanova)
- f180de333967f5d92306874f83628936ca732e1c Refactor getTableLayouts logic in AbstractTestHiveClient (Masha Basmanova)
- 564d288dc98bc02cc41cc45ccac488c4ce5501cf Change testBucketSortedTables to use ORC format (Masha Basmanova)
- 630a5924dd607ad142286754d69672904f3f7648 Add TestHiveUtils.getDefaultHiveSelectivePageSourceFactories (Masha Basmanova)
- fadcc9099d9f94bb8bb811f9593718c176c78361 Rename HiveTestUtils.getDefaultHiveDataStreamFactories (Masha Basmanova)
- 40f9db0f4762f1d623a6f399ccae0244cdefd40e Fix selective readers to make a copy of input positions (Bhavani Hari)
- b8acb68200b1cd90d809b63ac40871fed1cc08c6 Reset values after getBlock in LongSelectiveStreamReader (Bhavani Hari)
- 662a00a1780222fcffa2cc4de033bb2fbb82b506 Decode urlencoded property for PRESTO_EXTRA_CREDENTIAL parsing (Yi He)
- 323bd92e25bb00f467c915f43c2c1e046179269e Add Hive tests with filter pushdown and subfield pruning (Bhavani Hari)
- dcc5bb23eba4113b8ab9f3911c06aa7e7845bcfc Output verifier events for queries being skipped by pre-filters (Leiqing Cai)
- cad9d127fd6ce3d46b9bdee1276cd1a5b31f4858 Revert "Add Subfield to RowExpression converter" (Sahar Massachi)
- eac554cf72e9951170366942100f7347c2bd2fc7 Add Subfield to RowExpression converter (Sahar Massachi)
- 0637dccfadfe53fe8fdca1d21f629476a5825150 Revert "Add non-linear transformation for structural hash" (Wenlei Xie)
- d9c1e2084c70835d5feecf645ed8836064809fc2 fix organizationDiscoveryIntervalMillis bug (Ke Wang)
- f601571681050c5405227811bd18cbf798287923 Fix double and float NaN comparisons (Bhavani Hari)
- 22adf2280ad3c5ad1ca7b297a8fc494ac1567b12 Move TestHiveDistributedQueriesWithOptimizedRepartitioning to separate Travis Job (Bhavani Hari)
- e4769ad84577cb84355c5f1faae2c495d71d024f Add Hive smoke tests with filter pushdown and subfield pruning (Bhavani Hari)
- 89a41a9dd87baa3bc163da1bdbd4384a74311ee9 Allow creating flat-map reader for non-supported key types (Bhavani Hari)
- 1a6548649bb2f7dfaddb8e3d24f06dda56164078 Allow reading non-ORC tables when pushdown filter is enabled (Bhavani Hari)
- ffed56f03f43591a32a5ea77e4189c6cd8bbd1ae Fix GeometryToBingTiles for certain geometries (James A. Gill)
- 7018ccda951b050935bd9973d5f01b25f2f9084a Reduce the identifier length in generated code for nested columns (Venki Korukanti)
- 69c6573ba73afb205c84563ced29e9b590dbd841 Add expand_envelope function. (James A. Gill)
- 593e79048fbe011ed72f76819b09e1f4e4df5483 Allow empty Envelope serialization (James A. Gill)
- 60aa23d56f9388afed608363d902e22c05d1b8a6 Verify that outputPositions increase monotonically (Bhavani Hari)
- 2def2e13878b613523c91e60019b08b25fd7b402 Add JSON annotations to SqlFunctionId (Leiqing Cai)

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Release 0.230
+=============
+
+General Changes
+_______________
+* Fix an issue where a SQL function fails to compile when the function implementation contains constants.
+* Fix an issue where a SQL function fails to compile when some function parameters are not referenced in the function body.
+* Fix an issue where SQL functions do not to respect null-call clause during execution.
+* Fix compilation errors for expressions over types containing an extremely large number of nested types.
+* Fix a regression in lambda evaluation (Issue#13648).
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: shouldn't this be 

```
:issue: `13648`
```

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+* Add :func:`combinations` function, which returns n combinations of values in an array, up to ``n = 5``.
+* Add :func:`all_match()`, :func:`any_match()`, and :func:`none_match()` functions.
+* Add :func:`expand_envelope` function to return a geometry's envelope expanded by a distance.
+* Add population counts/weights to reservoir samples.
+* Improved ``PRESTO_EXTRA_CREDENTIAL`` header parsing to allow value contain multiple ``=`` characters and url-encoding characters.
+
+Hive Changes
+____________
+* Add Alluxio client jar to ``plugin/hive-hadoop2/`` (as a runtime dependency) to avoid having to copy the jar file to all Presto servers manually when connecting to Alluxio.
+* Improve ``BatchStreamReader`` performance.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: It is not clear if users understand what BatchStreamReader is. Perhaps, change to `Improve ORC reader performance.`

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+Raptor Changes
+______________
+* Fix an issue where organizer does not run scheduled job due to the misreading configuration property ``storage.organization-discovery-interval``.
+* Add ``table_supports_delta_delete`` table property in Raptor to allow deletions in the background.
+  ``DELETE`` queries in Raptor can now delete the data logically while relying on compactors to delete the data physically.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: - '*' is missing
- Perhaps, modify this sentence to start with a verb and remove `Raptor` (this whole section is about Raptor): `Modify DELETE queries to perform logical data deletion and rely on compactors to physically delete the data at a later point in time.`

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* Add support for Raptor to read data from HDFS while caching the files on local disks.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: - Is this enabled via a config/session property? If so, please, mention which one.
- Drop `Raptor` (this whole section is about Raptor): `Add support for caching data read from HDFS on a local disk`

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+* Add new cache module to allow using local disk for caching files from remote file systems.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: This is an implementation detail that users cannot understand. Let's remove.

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+Verifier Changes
+________________
+* Add skipped verification results to the output for queries being filtered.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: @caithagoras This sentence is hard to understand. Would you rephrase?

**Comment by highker** [comment]
Extra release note:

Add config `storage.zstd-jni-decompression-enabled` and `hive.zstd-jni-decompression-enabled` for Raptor and Hive connectors respectively to support JNI ZSTD decompressor for ORC files.

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+Hive Changes
+____________
+* Add Alluxio client jar to ``plugin/hive-hadoop2/`` (as a runtime dependency) to avoid having to copy the jar file to all Presto servers manually when connecting to Alluxio.
+* Improve ``BatchStreamReader`` performance.
+
+Raptor Changes
+______________
+* Fix an issue where organizer does not run scheduled job due to the misreading configuration property ``storage.organization-discovery-interval``.
+* Add ``table_supports_delta_delete`` table property in Raptor to allow deletions in the background.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: Is the property new or the capability new? If it is the latter (which seems to be the case to me) - we should say something like 

```
Add support for deleting data ..... . This can be enabled by adding the property ``table_supports...`` to the table property.
```

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 8]
```diff
@@ -0,0 +1,40 @@
+=============
+Release 0.230
+=============
+
+General Changes
+_______________
+* Fix an issue where a SQL function fails to compile when the function implementation contains constants.
+* Fix an issue where a SQL function fails to compile when some function parameters are not referenced in the function body.
+* Fix an issue where SQL functions do not to respect null-call clause during execution.
+* Fix compilation errors for expressions over types containing an extremely large number of nested types.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.230.rst]: For such vague ones, if it is not possible to clarify the fix, we should link to the PR/issue.

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 3 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: This is the same bullet point as the last line. @kewang1024 @highker Should we just remove this line?

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 7 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: Made some changes.
@kewang1024 @highker Could you confirm if the rephrase is ok? Is there a better rephase for "delta delete"?

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 6 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: Changed to
```
* Add support to always output verification results for queries being skipped for verification.
```

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 4 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: I think there are multiple configuration properties to control the behavior. @highker Could you confirm?

File: presto-docs/src/main/sphinx/release/release-0.230.rst [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+* Fix a regression in lambda evaluation (Issue#13648).
+* Fix :func:`geometry_to_bing_tile` function for geometries at -180 longitude or 85.05112878 latitude.
+* Add support to list non-builtin functions in SHOW FUNCTIONS. The feature can be turned on by the configuration property ``list-non-built-in-functions``.
+* Add support for ``DROP FUNCTION``.
+* Add :func:`combinations` function, which returns n combinations of values in an array, up to ``n = 5``.
+* Add :func:`all_match()`, :func:`any_match()`, and :func:`none_match()` functions.
+* Add :func:`expand_envelope` function to return a geometry's envelope expanded by a distance.
+* Add population counts/weights to reservoir samples.
```
[the remark this answers is not in the document]
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: Removed per @rschlussel 's comment: https://github.com/prestodb/presto/pull/13574#issuecomment-570354884

[the remark this answers is not in the document]
Comment (rschlussel) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: Thanks. Yeah, It's an internal refactoring so doesn't need a release note.

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 7 above)
Comment (kewang1024) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: So I think in current case, it's the first case actually, since we only merged the first PR into this release. So currently we just added new table property and haven't merged the capability yet.
Thus I think we should keep the original release note for now

Comment (mayankgarg1990) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: @kewang1024 - Should we publish both of these together, what is the value in announcing the property first?

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (yingsu00) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: @mayankgarg1990 @caithagoras ``BatchStreamReader`` is the legacy orc reader used in scan. Maybe we can rephrase it to "Improve orc reader performance"?

File: presto-docs/src/main/sphinx/release/release-0.230.rst (hunk 7 above)
Comment (kewang1024) [reply presto-docs/src/main/sphinx/release/release-0.230.rst]: Sounds ok to me, then should I remove it for now and add it in our next release?

**Comment by rschlussel** [comment]
travis failures are caused by  https://travis-ci.community/t/continuous-maven-repo-403/3908 / https://issues.sonatype.org/browse/MVNCENTRAL-4985.  The docs built successfully though, so I'll merge
